var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -227,6 +235,17 @@ def configure_args(self):
             if '+fortran' in self.spec:
                 extra_args.append('FC=%s' % self.spec['mpi'].mpifc)
 
+            # If using mpi, explicitly pass in compiler flags
+            CFLAGS.append(os.environ['CFLAGS'])
```

Comment:
same as https://github.com/spack/spack/pull/6716#discussion_r157633482

Reply:
I am likewise curious if the comment at https://github.com/spack/spack/pull/6716#discussion_r157634937 applies here.